docs/design.html

```diff
@@ -264,7 +264,8 @@ <h3 class="anchor-heading"><a id="semantics" class="anchor-link"></a><a href="#s
     can fail, cases where there are multiple consumer processes, or cases where data written to disk can be lost).
     <p>
     Kafka's semantics are straightforward. When publishing a message we have a notion of the message being "committed" to the log. Once a published message is committed, it will not be lost as long as one broker that
-    replicates the partition to which this message was written remains "alive". The definition of committed message and alive partition as well as a description of which types of failures we attempt to handle will be
+    replicates the partition to which this message was written remains "alive". More precisely, a message is considered committed only when all replicas in the in-sync replicas (ISR) for that partition have applied it to their log.
```

Comment:
This looks like a worthwhile improvement but I wonder whether swapping the sentences would make it even better. So I mean:

`Kafka's semantics are straightforward. When publishing a message we have a notion of the message being "committed" to the log. A message is considered committed only when all replicas in the in-sync replicas (ISR) for that partition have applied it to their log. Once a published message is committed, it will not be lost as long as one broker that replicates the partition to which this message was written remains "alive". The definition of committed message and alive partition as well as a description of which types of failures we attempt to handle will be described in more detail in the next section.`